src/providers/ipa/ipa_s2n_exop.c

```diff
@@ -2082,15 +2083,21 @@ static errno_t get_groups_dns(TALLOC_CTX *mem_ctx, struct sss_domain_info *dom,
             goto done;
         }
 
-        /* This might fail if some unexpected cases are used. But current
-         * sysdb code which handles group membership constructs DNs this way
-         * as well, IPA names are lowercased and AD names by default will be
-         * lowercased as well. If there are really use-cases which cause an
-         * issue here, sysdb_group_strdn() has to be replaced by a proper
-         * search. */
-        dn_list[c] = sysdb_group_strdn(dn_list, dom->name, name_list[c]);
+        /* If the group name is overridden in the default view we have to
+         * search for the name and cannot construct it because the extdom
+         * plugin will return the overridden name but the DN of the related
+         * group object in the cache will contain the original name. */
+
+        ret = sysdb_search_group_by_name(tmp_ctx, dom, name_list[c], NULL,
+                                         &msg);
+        if (ret == EOK) {
+            dn_list[c] = ldb_dn_alloc_linearized(dn_list, msg->dn);
+        } else {
+            /* best effort, try to construct the DN */
+            dn_list[c] = sysdb_group_strdn(dn_list, dom->name, name_list[c]);
```

Comment:
Isn't worth to add a debug message here (at SSSDBG_TRACE_FUNC level) letting us know that we're building the DN? Wouldn't it make our lives easier in the future by knowing that we're reaching this code path?

If not, please, just ignore the comment.

Reply:
That's a good idea, comment added.